### PR TITLE
Using UIImageView instead of BPKIconView for airline logo

### PR DESCRIPTION
**Pull request by JLau-Sky**
Fixed BPKFlightLeg to take an UIImage instead of an icon name for airline logos.

- Example updated
- Snapshots updated

### CODE REVIEW

File: Backpack/FlightLeg/Classes/BPKFlightLeg.swift [hunk 1]
```diff
@@ -146,15 +146,18 @@ public final class BPKFlightLeg: UIView {
...
         
         NSLayoutConstraint.activate([
             leadingStackView.topAnchor.constraint(equalTo: verticalStackView.topAnchor, constant: BPKSpacingSm),
             leadingStackView.bottomAnchor.constraint(equalTo: verticalStackView.bottomAnchor),
             leadingStackView.leadingAnchor.constraint(equalTo: verticalStackView.leadingAnchor),
-            leadingStackView.trailingAnchor.constraint(equalTo: verticalStackView.trailingAnchor)
+            leadingStackView.trailingAnchor.constraint(equalTo: verticalStackView.trailingAnchor),
+            
+            carrierLogoIcon.widthAnchor.constraint(equalToConstant: 24),
+            carrierLogoIcon.heightAnchor.constraint(equalToConstant: 24)
```
Comment (gert-janvercauteren) [inline Backpack/FlightLeg/Classes/BPKFlightLeg.swift:160]: Can we use BPKSpacings for these 'magic' numbers. A constant defined at the top is also ok.

File: Backpack/Tests/SnapshotTests/BPKFlightLegSnapshotTest.swift [hunk 2]
```diff
@@ -47,7 +47,7 @@ final class BPKFlightLegSnapshotTest: XCTestCase {
         duration: String = "7h 55m",
         operatedBy: String? = nil,
         warning: String? = nil,
-        carrierLogo: BPKIconName = .airline
+        carrierLogo: UIImage? = BPKIcon.makeLargeIcon(name: .airline, color: .bpk_black)
```
Comment (gert-janvercauteren) [inline Backpack/Tests/SnapshotTests/BPKFlightLegSnapshotTest.swift:50]: Could we use BPKColor.textOnLight for this? Or (worst case) BPKColor.black? 
We want to reduce the amount of `bpk_` usages.

*Same for the below usages

File: Example/Backpack/UIKit/Components/Flight Leg/BPKFlightLegViewController.swift [hunk 3]
```diff
@@ -103,7 +103,7 @@ class BPKFlightLegViewController: UIViewController {
             flightDescription: defaultDescription,
             stopsInfo: "Direct",
             duration: "7h 55m",
-            carrierLogo: .airline
+            carrierLogo: BPKIcon.makeLargeIcon(name: .airline, color: .bpk_black)
```
Comment (gert-janvercauteren) [inline Example/Backpack/UIKit/Components/Flight Leg/BPKFlightLegViewController.swift:106]: SImilar to above regarding color token usage. (as below)

**Review by gert-janvercauteren — COMMENTED** [review]
Changes look good, some tiny nitpicks.
Did you make the snapshots using Rosetta? I can't discern any changes compared to the originals

**Comment by JLau-Sky** [comment]
> 

Nope it's not Rosetta, Xcode 14.2 iPhone SE 3rd Gen 16.2

**Review by gert-janvercauteren — APPROVED** [review]

**Pull request merged**